packages/server/graphql/mutations/updatePokerTemplateScaleValue.ts

```diff
@@ -0,0 +1,94 @@
+import {GraphQLID, GraphQLInt, GraphQLNonNull} from 'graphql'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import getRethink from '../../database/rethinkDriver'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import publish from '../../utils/publish'
+import standardError from '../../utils/standardError'
+import TemplateScaleInput from '../types/TemplateScaleInput'
+import UpdatePokerTemplateScaleValuePayload from '../types/UpdatePokerTemplateScaleValuePayload'
+import {
+  validateColorValue,
+  validateScaleLabel,
+  validateScaleValue
+} from './helpers/validateScaleValue'
+
+const updatePokerTemplateScaleValue = {
+  description: 'Update a scale value for a scale in a poker template',
+  type: UpdatePokerTemplateScaleValuePayload,
+  args: {
+    scaleId: {
+      type: new GraphQLNonNull(GraphQLID)
+    },
+    scaleValue: {
+      type: new GraphQLNonNull(TemplateScaleInput)
+    },
+    index: {
```

Comment:
we can use `beforeValue` instead of index to handle drag & drop behavior.

for non-numeric scales, we should update the values going from least to greatest.
e.g. `[{value: 0, label: 'S'}, {value: 1, label: 'M'}]` if i drag the M above the S, then it should read:
`[{value: 0, label: 'M', {value: 1, label: 'S'}]`
the args would be:
```js
scaleId: 'foo',
scaleValue: 1,
beforeValue: 0
```

Reply:
beforeValue is also susceptible to race conditions, so index works here

Reply:
@tianrunhe

Reply:
@mattkrick I have deeper thinking about this today. I just felt that our data model is fragile. So our current thinking is that a scale value has a label and a numerical value. We give the user the ability to edit the label and then we "infer" the numerical value on our end. Think about the following cases, because hey users can put whatever they want:  
  
Edge case 1: user creates three scale values, in the order, "1", "S", "M", "2". How do we assign numerical value for each label here?

Even more extreme cases: user creates three scale values, in the order, "2", "3", "1".  Do we wanna "be smart" and say that oh I think you actually mean 1, 2, 3, or do we respect the ordering user put in, and then weirdly map them to the ordering?

So one solution is that when create a scale, we ask would you like a label only scale or a numerical scale? Then for label only scale, we respect the ordering they put in. For numerical scale, we put a client side check for sth like "hey you cannot move a 3 in the middle of 1 and 2".

Feels like more work either way. Any simple solutions here?

Reply:
idea 1: forced sort. if when they add a new value, we pick where it goes. we sort all numeric values from least to greatest. all non-numeric values come after the numeric values. they have free rein over how non-numeric values are sorted. this is bad because it may feel too magical.

idea 2:  when the label is non-numeric, a numeric "value" input field appears. basically, we have a hidden "value" column & when 1 or more labels are non-numeric, that column gets unhidden @ackernaut for input & design. i think this is probably the best.

idea 3: throw an error. if they enter a non-numeric value in a numeric set, we say they can't do that. this limits functionality, so it's not great.